File: 0383 ransom-note/solution.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
       int arr[] = new int[26];
       char[] cr=magazine.toCharArray();
       char[] rr = ransomNote.toCharArray();
       for(char c:cr) arr[c-'a']++;
       for(char c:rr) {
         arr[c-'a']--;
         if(arr[c-'a']<0) return false;
       }
       return true;
    }
}

File: 0383 ransom-note/solution.go
func canConstruct(ransomNote string, magazine string) bool {
    var arr [26]int

    for _,v := range magazine {
        arr[v-'a']++
    }
    for _,v := range ransomNote {
        arr[v-'a']--
        if arr[v-'a']<0 {
            return false
        }
    }
    return true
}
